include/pybind11/pybind11.h

```diff
@@ -223,20 +231,48 @@ class cpp_function : public function {
         }
     }
 
+    // Utility class that keeps track of all duplicated strings, and cleans them up in its destructor,
+    // unless they are released. Basically a RAII-solution to deal with exceptions along the way.
+    class strdup_guard {
+    public:
+        ~strdup_guard() {
+            for (auto s : strings)
+                std::free(s);
+        }
+        char *operator()(const char *s) {
+            auto t = strdup(s);
+            strings.emplace_back(t);
+            return t;
+        }
+        void release() {
+            strings.clear();
```

Comment:
`strings.clear()` doesn't call `free()` on the cleared pointers. Should it?

Reply:
Nope, or it wouldn't be releasing! (This name is meant to mirror `unique_ptr::release()`.)